src/common/ceph_context.h

```diff
@@ -79,11 +81,40 @@ class CephContext {
   ceph::common::PerfCountersCollection& _perf_counters_collection;
   CephContext* get();
   void put();
+
+  /**
+   * Get the admin socket associated with this CephContext.
+   *
+   * Note that an admin socket should only be created for
+   * daemons.
+   *
+   * @return the admin socket or nullptr
+   */
+  AdminSocket* get_admin_socket() { 
```

Comment:
In a vacuum, I'd assume that with this signature get_admin_socket() gives me a pointer to an AdminSocket that I own and am responsible for destructing.  If the intention is for the consumer not to take ownership, this should probably be a std::optional<AdminSocket&> or it should assert that asok exists.

Reply:
OK. I agree with your reasoning in this case. Will change.
Note though that generally I only consider returning a unique_ptr as a strong hint of passing
ownership.

Reply:
Sorry, I just noticed that you did this for uniformity with the existing interface, that's reasonable.